lib/rules/no-implicit-coercion.js

```diff
@@ -299,6 +304,33 @@ module.exports = {
 
                     report(node, recommendation, true);
                 }
+            },
+
+            TemplateLiteral(node) {
+                if (!options.templateString) {
+                    return;
+                }
```

Comment:
If this option should apply even if `string` is set to `false`, then we should add a note about that to the documentation.